packages/cli/test/ci-collect.test.js

```diff
@@ -0,0 +1,117 @@
+const {runCLI} = require('./test-utils.js');
+
+describe('collect', () => {
+  const fixturesDir = path.join(__dirname, 'fixtures');
+  const rcFile = path.join(fixturesDir, 'lighthouserc.json');
+
+  let tempDir;
+  beforeAll(() => {
+    tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'lighthouserc-'));
+  });
+
+  it('should collect results from staticDistDir', () => {
+    const {stdout, stderr, status} = runCLI(
+      ['collect', `--config=${rcFile}`, `--static-dist-dir=${fixturesDir}`],
+      {
+        // Run in temp dir to avoid conflicts with other tests
+        cwd: tempDir,
+      }
+    );
+
+    const stdoutClean = stdout;
+    expect(stdoutClean).toMatchInlineSnapshot(`
+      "Started a web server on port XXXX...
+      Running Lighthouse 2 time(s) on http://localhost:XXXX/checkout.html
+      Run #1...done.
+      Run #2...done.
+      Running Lighthouse 2 time(s) on http://localhost:XXXX/index.html
+      Run #1...done.
+      Run #2...done.
+      Done running Lighthouse!
+      "
+    `);
+    expect(stderr.toString()).toMatchInlineSnapshot(`""`);
+    expect(status).toEqual(0);
+  }, 90000);
+
+  it('should collect results with a server command with custom start pattern', () => {
+    // FIXME: for some inexplicable reason this test cannot pass in Travis Windows
+    if (os.platform() === 'win32') return;
+
+    const serverPath = path.join(fixturesDir, 'autorun-start-server/autorun-server.js');
+    const startCommand = `SERVER_START_MESSAGE='Running server' node ${serverPath}`;
+    const {stdout, stderr, status} = runCLI(
+      [
+        'collect',
+        `-n=1`,
+        `--config=${rcFile}`,
+        `--start-server-command=${startCommand}`,
+        '--start-server-ready-pattern=running',
+        '--url=http://localhost:52425/',
+      ],
+      {
+        // Run in temp dir to avoid conflicts with other tests
+        cwd: tempDir,
+      }
+    );
+
+    // Check server started and lighthouse ran.
+    expect(stdout).toMatchInlineSnapshot(`
+      "Started a web server with \\"${startCommand}\\"...
+      Running Lighthouse 1 time(s) on http://localhost:XXXX/
+      Run #1...done.
+      Done running Lighthouse!
+      "
+    `);
+    // Checkout no errors were logged.
+    expect(stderr.toString()).toMatchInlineSnapshot(`""`);
+    // Check script ran without errors.
+    expect(status).toEqual(0);
+  });
+
+  it('should print timeout message for server command not printing a matchable pattern', () => {
+    // FIXME: for some inexplicable reason this test cannot pass in Travis Windows
+    if (os.platform() === 'win32') return;
+
+    const serverPath = path.join(fixturesDir, 'autorun-start-server/autorun-server.js');
+    const startCommand = `SERVER_START_MESSAGE='Running server' node ${serverPath}`;
+    const {stdout, status} = runCLI(
+      [
+        'collect',
+        `-n=1`,
+        `--config=${rcFile}`,
+        `--start-server-command=${startCommand}`,
+        '--url=http://localhost:52425/',
```

Comment:
these have the same port that seems like a problem, maybe make it another env variable?

Reply:
🤔 why is that a problem? From what I see the server is stopped after the test or we would be getting a _something is already running in that port_ error.

Reply:
when everything goes as expected it shouldn't :) but we've seen flaky tests in lighthouse-core when some race means the port isn't released back into the os pool on CI machines fast enough and so the next one will flakily fail, just easier to use unique ports, plus it means we can run these in parallel in the future since we're already doing some of the split work :)

Reply:
Got it. 

As the lib grows we should consider a solution like: https://www.npmjs.com/package/get-port. 

It can get hard to keep an eye on every port to make sure we are not repeating them.

Reply:
Ironically it's the exact same method that `get-port` uses that we've seen get us into trouble on super slow machines (opening up and quickly closing a server on a port) 😆but point taken, yes a util for generating unique ports would be helpful

Reply:
🤣 what a small world.